### PR TITLE
Update travis scripts to test quickstart over  jdk 10-15

### PR DESCRIPTION
**Pull request by fx19880617**
- Adding More JDK support(JDK 8 and 10-15) for QuickStart.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Can you please describe the change for the testing? Or should that be separate into another pr?

**Comment by fx19880617** [comment]
> Can you please describe the change for the testing? Or should that be separate into another pr?

Updated. It's pretty minor, I feel it's ok to merge them into one PR

File: .travis_quickstart.sh [hunk 1]
```diff
@@ -50,9 +50,9 @@ bin/quick-start-streaming.sh &
...
 sleep 60
 COUNT_STAR_RES_1=`curl -X POST --header 'Accept: application/json'  -d '{"sql":"select count(*) from meetupRsvp limit 1","trace":false}' http://localhost:8000/query/sql | jq '.resultTable.rows[0][0]'`
-sleep 15
+sleep 30
 COUNT_STAR_RES_2=`curl -X POST --header 'Accept: application/json'  -d '{"sql":"select count(*) from meetupRsvp limit 1","trace":false}' http://localhost:8000/query/sql | jq '.resultTable.rows[0][0]'`
-sleep 15
+sleep 30
 COUNT_STAR_RES_3=`curl -X POST --header 'Accept: application/json'  -d '{"sql":"select count(*) from meetupRsvp limit 1","trace":false}' http://localhost:8000/query/sql | jq '.resultTable.rows[0][0]'`
 if [ "${COUNT_STAR_RES_3}" -le "${COUNT_STAR_RES_2}" ] || [ "${COUNT_STAR_RES_2}" -le "${COUNT_STAR_RES_1}" ]; then
   echo 'Streaming Quickstart: Not getting incremental counts for 3 consecutive count star queries with 15 seconds interval.'
```
Comment (Jackie-Jiang) [inline .travis_quickstart.sh]: Update this line to 30 seconds

File: .travis_quickstart.sh [hunk 2]
```diff
@@ -95,9 +95,9 @@ sleep 30
...
```
Comment (Jackie-Jiang) [inline .travis_quickstart.sh]: Same here

File: .travis.yml [hunk 3]
```diff
@@ -39,19 +39,20 @@ jobs:
         - ./.travis_test.sh
       env:
         - RUN_INTEGRATION_TESTS=false
-    - name: "QuickStart - OracleJDK 8"
+    - name: "QuickStart - Java 8 & OpenJDK 14-15"
```
Comment (Jackie-Jiang) [inline .travis.yml]: Why splitting the tasks this way? How about the other OracleJDK versions?

File: .gitignore [hunk 4]
```diff
@@ -21,4 +21,5 @@ docs/_build
 .*.swp
 **/.docker/
 **/.env
+**/.factorypath
```
Comment (Jackie-Jiang) [inline .gitignore]: Where is this created?

File: .travis_quickstart.sh [hunk 5]
```diff
@@ -36,90 +39,98 @@ cd ${DIST_BIN_DIR}
 # Test quick-start-batch
 bin/quick-start-batch.sh &
 PID=$!
-sleep 60
-COUNT_STAR_RES=`curl -X POST --header 'Accept: application/json'  -d '{"sql":"select count(*) from baseballStats limit 1","trace":false}' http://localhost:8000/query/sql | jq '.resultTable.rows[0][0]'`
-if [ "${COUNT_STAR_RES}" -ne 97889 ]; then
-  echo 'Batch Quickstart: Incorrect result for count star query.'
+
+PASS=0
+sleep 30
```
Comment (Jackie-Jiang) [inline .travis_quickstart.sh]: Can we avoid this sleep if possible and retry before the cluster start responding? Same for real-time and hybrid

File: .travis_quickstart.sh [hunk 6]
```diff
@@ -36,90 +39,98 @@ cd ${DIST_BIN_DIR}
 # Test quick-start-batch
 bin/quick-start-batch.sh &
```
Comment (Jackie-Jiang) [inline .travis_quickstart.sh]: Do you want to use one of the quick-starts for the relative path test?

File: .travis_quickstart_openjdk.sh [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+rm -rf "${JAVA_HOME}"
+. ~/bin/install-jdk.sh --feature "${JDK_VERSION}" --cacerts --target "${JAVA_HOME}"
+gdb --batch-silent --pid=$$ --eval-command='call unbind_variable("script_name")'
+gdb --batch-silent --pid=$$ --eval-command='call unbind_variable("script_version")'
+rm -rf jdk.tar.gz
+echo "${JAVA_HOME}"
+java -version
+./.travis_install.sh
+./.travis_quickstart.sh
+rm -rf "${JAVA_HOME}"
```
Comment (Jackie-Jiang) [inline .travis_quickstart_openjdk.sh]: (nit) Add a line

File: .gitignore (hunk 4 above)
Comment (fx19880617) [reply .gitignore]: I somehow got it when update Maven Project using IDE Maven plugins

File: .travis.yml (hunk 3 above)
Comment (fx19880617) [reply .travis.yml]: Just don't want to have many parallel Quickstart jobs.
Also tries to keep the test time of one job to not exceed unit test/integration test.

File: .travis_quickstart.sh (hunk 5 above)
Comment (fx19880617) [reply .travis_quickstart.sh]: I've test it out, it requires more than 30 secs to bring up quickstart ready for query. if we removed this line, we will see more garbage logs of curl failures

File: .travis_quickstart.sh (hunk 6 above)
Comment (fx19880617) [reply .travis_quickstart.sh]: hybrid test is for relative path

File: .travis_quickstart.sh (hunk 5 above)
Comment (fx19880617) [reply .travis_quickstart.sh]: and hybrid quickstart requires about 1 minute to start :(

**Pull request merged**